capture.js

```diff
@@ -0,0 +1,47 @@
+const capture = {
+
+  init() {
+    this.addListeners();
+  },
+
+  addListeners() {
+    // listen for each HTTP response
+    browser.webRequest.onResponseStarted.addListener(this.collectRequests, {urls: ['<all_urls>']}
+    );
+  },
+
+  /*
+  * makes calls to the storage API for each
+  * first and third party request
+  */
+  async collectRequests(response) {
+    const tab = await browser.tabs.get(response.tabId);
+    let tabUrl = new URL(tab.url);
+    tabUrl = tabUrl.hostname;
+    let targetUrl = new URL(response.url);
+    targetUrl = targetUrl.hostname;
+    let originUrl = new URL(response.originUrl);
+    originUrl = originUrl.hostname;
+
+    //capture first party requests
```

Comment:
nit: space before `//` there is an ESLint rule for this to enforce consistency (when I say nit it is to be taken as, hey I'm pedantic and I'm sorry about that)